### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by resir014**
**Work in progress. Do not merge!**

This is a master PR to track all the changes made for version 2.0.0 of `typed-screeps`, where we'll be bringing a host of updates to the declaration files.

## Ship List

* [x] Add generic overloads for `.find()` functions (#14)
* [x] Fix some of the TODOs in `constants.ts`
* [x] Project files restructuring
* [x] Finish writing test cases for in-game APIs
* [x] Add missing in-game APIs (if any)

## Chores

* [x] Set up Travis CI config
* [x] Add community profile stuff (CoC, licence, issue & PR templates)

More will be added soon.

### CODE REVIEW

**Comment by Arcath** [comment]
Should we work on our GitHub Community Profile?

We still need:

 - [ ] Code of Conduct
 - [ ] Contributing guide
 - [ ] License
 - [ ] Issue & PR Template

**Comment by resir014** [comment]
@Arcath Yeah, sure!

We're MIT, but I'm still not sure who to assign the copyright to, since this project has changed a couple of times already.

I can include a Code of Conduct and a basic contributing guidelines tomorrow.

File: README.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ Installing through `@types/` will be made available soon. Hang tight!
 
 ## Contribute
 
-Issues and Pull Requests are welcom! Please read the [Contributing Guidelines](.github/CONTRIBUTING.md) and [Code of Conduct](.github/CODE_OF_CONDUCT.md) beforehand.
+Issues and Pull Requests are welcome! Please read the [Contributing Guidelines](.github/CONTRIBUTING.md) and [Code of Conduct](.github/CODE_OF_CONDUCT.md) beforehand.
```
Comment (apemanzilla) [inline README.md]: :+1:

**Comment by Arcath** [comment]
I need to make a change to my last PR. It fixes the problem but adds a small bug.

`find<StructureTower>` causes the typings to become `{}` for the `FindConstant` it doesn't break anything but I would prefer to get it working so that:

```ts
let towers = room.find<StructureTower>(FIND_CREEPS)
```

causes an error as `FIND_CREEPS` can't return anything that can satisfy `StructureTower`

**Comment by resir014** [comment]
Alright, looks like we're merging this then.

Thanks @Arcath @apemanzilla!

**Pull request merged**